### DISCUSSION TITLE
How to specify docker run options when creating a generic container

### DISCUSSION THREAD

**Discussion by r3dfur** [testcontainers/testcontainers-java#5481] [Q&A]
i'm working with  [atmoz/sftp](https://hub.docker.com/r/atmoz/sftp)
I would mount my own host keys as described:

docker run \
    -v <host-dir>/id_rsa.pub:/home/foo/.ssh/keys/id_rsa.pub:ro \
    -v <host-dir>/id_other.pub:/home/foo/.ssh/keys/id_other.pub:ro \
    -v <host-dir>/share:/home/foo/share \
    -p 2222:22 -d atmoz/sftp \
    foo::1001

i try with:

```
    private static GenericContainer buildSFTPContainer(Path filePath) {
        return new GenericContainer<>(DockerImageName.parse("atmoz/sftp:latest"))
                .withCreateContainerCmdModifier(createContainerCmd -> {
                    createContainerCmd.withCmd("-v /tmp/keys/sftp_rsa_key:/etc/ssh/sftp_rsa_key ");
                })
                .withExposedPorts(PORT)
                .withCopyFileToContainer(MountableFile.forHostPath(filePath),
                        "/home/" + USER + REMOTE_PATH + filePath.getFileName())
                .withCommand(USER + ":1001:::upload");


    }
```
but it doesn't work.

help!!!
Thanks

**Comment by kiview** [comment]
You are modifying the Docker `CMD` in your code snippet. There should be no need to modify `CMD` or `ENTRYPOINT`. You should instead only call `withCopyFileToContainer` multiple times, for each of your files. 

The code replicating your `docker run` could look something like this (not tested and won't probably work verbatim):

```java
return new GenericContainer<>(DockerImageName.parse("atmoz/sftp:latest"))
        .withExposedPorts(22) // this will map to a dynamic port on the host
        .withCopyFileToContainer(MountableFile.forHostPath("/id_rsa.pub"),
                "/home/foo/.ssh/keys/id_rsa.pub")
        .withCopyFileToContainer(MountableFile.forHostPath("/id_other.pub"),
                "/home/foo/.ssh/keys/id_rsa.pub")
        .withCopyFileToContainer(MountableFile.forHostPath("/share"), // this won't sync the files from within the container back to the host 
                "/home/foo/share") 
        .withCommand("foo::1001");
```

**Reply by WrettySV** [reply]
Sorry, why only pub key is copied to the container? What about private key? 
I have done the same but with private and public keys and there is always the same mistake either com.jcraft.jsch.JSchException: Algorithm negotiation fail, or com.jcraft.jsch.JSchException: Auth fail

**Reply by anujpradhaan** [reply]
That's because that's how SSH works. 

![image](https://github.com/testcontainers/testcontainers-java/assets/5353322/e0936ff2-a28d-4064-8735-5a79669e1f21)

Private key is specific to you only. Sharing that as well is not much of a security. I would suggest reading basics of ssh.
